### ISSUE TITLE
Windows build

### ISSUE THREAD

**Issue by chaplin89** [samtools/samtools#2064]
#### Is your feature request related to a problem? Please specify.
No.

#### Describe the solution you would like.
Hi,

without a better alternative to integrate samtools/htslib/bcftools (and a bunch of other stuff) in my Python/Windows project, I've built a GitHub action to compile these tools under msys2 and build a wheel with them to upload to PyPI. It's far from being ideal but I'm not sure what alternatives do I have here since pysam don't seem to support Windows for the moment.

Regardless, I think that this GH action can benefit also samtools/htslib/bcftools in case you want a Windows build. It's nothing complicated and I think it's a very low hanging fruit that will allow to have a windows release with a very low effort. You can see it in action here: https://github.com/WGSE-NG/External/blob/main/.github/workflows/compile-and-publish.yml

I can test it also on this repo and open a CR if there's some interest from your side.

Marco

**Comment by jkbonfield** [comment]
I think publishing python wheels is outside of our remit and should be an issue raised elsewhere.  We're not supporting the interaction with other languages.  Maybe it's something to raise with PySAM?

The issue of prebuilt binaries has been raised before though.  There are easy solutions on other platforms (eg homebrew or Conda), but I'm not sure if the windows community has a similar alternative.  I suspect it's things like the prevalence conda which persuaded us against taking on the maintenance burden of maintaining our own set of binaries.

**Comment by chaplin89** [comment]
>I think publishing python wheels is outside of our remit and should be an issue raised elsewhere. We're not supporting the interaction with other languages. Maybe it's something to raise with PySAM?

Sorry I was really not suggesting you should release wheels containing exes. I think it's a very poor solution and I'm doing that just because I don't think I have other choices. I added that part just to explain what lead me to this route.

> Edit: actually there is conda for Windows

I was suggesting it because it seems you already have binaries for linux built in CI and adding the windows one seems a very low effort thing to do, especially since the build process will be exactly the same. Conda for windows is a good option I guess, but my tool is made for laymen (as I am) and I would prefer to stay away from conda unless I'm really forced to use it. In any case is fine, I can keep using my version if it's something you have no interest.

**Comment by jkbonfield** [comment]
Thank you for the pointers on GitHub Actions.  Given the relatively slow start up of AppVeyor I explored using GitHub Actions to build Windows and after a few odd teething problems it's now working.  A variety of PRs now exist, or have been merged.  See https://github.com/samtools/htslib/pull/1796 and https://github.com/samtools/samtools/pull/2072.

So I'm closing this now as we don't intend to go down the binary release publishing avenue for now, but the CI components was helpful.

Thanks
